src/AutoRest.CSharp/LowLevel/Output/Tests/DpgClientTestProvider.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using AutoRest.CSharp.Common.Output.Expressions.Statements;
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
+using AutoRest.CSharp.Common.Output.Models;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Input.Source;
+using AutoRest.CSharp.LowLevel.Output.Samples;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Samples.Models;
+using NUnit.Framework;
+
+namespace AutoRest.CSharp.LowLevel.Output.Tests
+{
+    internal class DpgClientTestProvider : DpgClientSampleProvider
```

Comment:
In my opinion, the test provider and the sample provider does not has the inheritance relationship. instead, the test provider and sample provider has many common stuffs. So maybe it is better to define a base class to contain the common stuff, and let test provider and sample provider inheritance the base class.

Reply:
I do not have any ideas of how to name that common base type. Any suggestions?

Reply:
Thank you!
Well as for the renaming and moving to common, we could do it later when we find it is necessary

Reply:
Done

Reply:
Thanks. in the `DPGClientTestProvider` there are remaining some variable/type suffixed with `Sample`. We can refact this later, current it is ok. resolve this.